src/main/java/oof/Storage.java

```diff
@@ -114,44 +132,20 @@ public void writeTaskList(TaskList taskList) {
      * @throws IOException if file does not exist.
      */
     public ArrayList<Task> readTaskList() throws IOException {
-        String filename = "output.txt";
-        BufferedReader reader = new BufferedReader(new FileReader(filename));
+        BufferedReader reader = new BufferedReader(new FileReader(PATH_OUTPUT));
         String line;
         int counter = 0;
         while ((line = reader.readLine()) != null) {
             counter += 1;
+            String[] lineSplit = line.split(DELIMITER_ESCAPED);
             if (isTodo(line)) {
-                String details = line.substring(INDEX_LABEL_END).trim();
-                String[] lineSplit = details.split("\\(on: ");
-                String description = lineSplit[INDEX_DESCRIPTION].trim();
-                String on = lineSplit[INDEX_DATE].trim().substring(0, lineSplit[INDEX_DATE].length() - 1);
-                Todo todo = new Todo(description, on);
-                arr.add(todo);
-                if (checkDone(line)) {
-                    arr.get(counter - 1).setStatus();
-                }
+                addToDo(lineSplit, counter);
             } else if (isDeadline(line)) {
-                String details = line.substring(INDEX_LABEL_END).trim();
-                String[] lineSplit = details.split("\\(by: ");
-                String description = lineSplit[INDEX_DESCRIPTION].trim();
-                String date = lineSplit[INDEX_DATE].trim().substring(0, lineSplit[INDEX_DATE].length() - 1);
-                Deadline deadline = new Deadline(description, date);
-                arr.add(deadline);
-                if (checkDone(line)) {
-                    arr.get(counter - 1).setStatus();
-                }
+                addDeadline(lineSplit, counter);
             } else if (isEvent(line)) {
-                String details = line.substring(INDEX_LABEL_END).trim();
-                String[] lineSplit = details.split(" \\(from: ");
-                String description = lineSplit[INDEX_DESCRIPTION].trim();
-                String[] dateSplit = lineSplit[INDEX_DATE].split("to: ");
-                String startDate = dateSplit[INDEX_START].trim().substring(0, dateSplit[INDEX_START].length() - 1);
-                String endDate = dateSplit[INDEX_END].trim().substring(0, dateSplit[INDEX_END].length() - 1);
-                Event event = new Event(description, startDate, endDate);
-                arr.add(event);
-                if (checkDone(line)) {
-                    arr.get(counter - 1).setStatus();
-                }
+                addEvent(lineSplit, counter);
+            } else if (isAssignment(line)) {
+                addAssignment(lineSplit, counter);
             }
```

Comment:
Since we implemented parsing of save file my delimiters, this entire section can be improved by using switch case on lineSplit[0]